packages/dds/merge-tree/src/segmentGroupCollection.ts

```diff
@@ -21,9 +21,16 @@ export class SegmentGroupCollection {
         return this.segmentGroups.empty();
     }
 
-    public enqueue(segmentGroup: SegmentGroup) {
+    public enqueue(segmentGroup: SegmentGroup, sourceSegment?: ISegment) {
         this.segmentGroups.enqueue(segmentGroup);
         segmentGroup.segments.push(this.segment);
+        if (segmentGroup.previousProps && sourceSegment) {
+            // duplicate the previousProps for the new segment if it's split from an existing one
+            const index = segmentGroup.segments.indexOf(sourceSegment);
+            if (index !== -1) {
+                segmentGroup.previousProps.push(segmentGroup.previousProps[index]);
```

Comment:
Is it ok to push a reference here instead of a deep copy? No risk of the previous props being updated in one of the two segments but getting changed in both?

Reply:
i think this should be fine, as previousProps are created via the PropertyDeltas, so the segment no longer uses or need them, so they shoul never be modified.  fewer object allocations are better if we can avoid it

Reply:
previousProps never updates. It contains the properties before the edit that created the segmentGroup, so changing it would mean somehow changing history.